client/src/main/java/io/pravega/client/connection/impl/ConnectionPoolImpl.java

```diff
@@ -194,6 +195,7 @@ public void pruneUnusedConnections() {
     }
 
     @Override
+    @Synchronized
```

Comment:
Did not understand the reason for this change here.

Reply:
Removed